### ISSUE TITLE
I can not send email from icloud.com email account

### ISSUE THREAD

**Issue by potetochippusu** [MailCore/mailcore2#1815]
[Question]
Did MailCore2 stop sending emails for icloud.com accounts?

[Development environment]
MailCore2 ver.0.6.4
iOS 12.2 (iPhone 5S)
Swift 4.2

[phenomenon]
When sending an email using an icloud.com account in the above environment,
No errors occur, but no emails to the destination.

However, the emulator will receive an email.
When I use a real iPhone, I can not receive an email.
You can also send it using an email account other than icloud.com.

There is no record received in the destination mail server log, so
Is there anything that is blocked by icloud's server?

**Comment by haithngn** [comment]
Could you please show us the SMTP configuration you used to send message and the SMTP logs ?

**Comment by potetochippusu** [comment]
Here you will.
Is that caused by having the same iCloud email account settings on the device mailer?

[Source code]
-------------------------------------------------------------------
let smtpSession = MCOSMTPSession()

smtpSession.hostname = "smtp.mail.me.com"
smtpSession.username = "[ICLOUD_MAILADDRSS]"
smtpSession.password = "[AppPassword]"
smtpSession.port = 587
smtpSession.authType = MCOAuthType.saslLogin
smtpSession.connectionType = MCOConnectionType.startTLS

smtpSession.connectionLogger = {(connectionID, type, data) in
    if data != nil {
        if let string = NSString(data: data!, encoding: String.Encoding.utf8.rawValue){
            NSLog("Connectionlogger: \(string)")
        }
    }
}

let builder = MCOMessageBuilder()
builder.header.to = [MCOAddress(displayName: "", mailbox: "[TO_ADDRESS]")]
builder.header.from = MCOAddress(displayName: "", mailbox: "[ICLOUD_MAILADDRSS]")

builder.header.subject = "[MAIL_TITLE]"
builder.htmlBody = "[MAIL_BODY]"

let rfc822Data = builder.data()
let sendOperation = smtpSession.sendOperation(with: rfc822Data)
sendOperation!.start { (error) -> Void in
    if (error != nil) {
        NSLog("Error sending email: \(error)")
    } else {
        NSLog("Successfully sent email!")
    }
}
-------------------------------------------------------------------

[Send log]
-------------------------------------------------------------------
2019-05-29 11:44:26.349089+0900 [PRJ_NAME][309:8621] Connectionlogger: 220 pv50p00im-zteg10021401.me.com 3.3.1 v.666
2019-05-29 11:44:26.349777+0900 [PRJ_NAME][309:8621] Connectionlogger: EHLO [iPhone_NAME]
2019-05-29 11:44:26.492623+0900 [PRJ_NAME][309:8621] Connectionlogger: 250-pv50p00im-zteg10021401.me.com
250-PIPELINING
250-SIZE 28311552
250-ETRN
250-STARTTLS
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 DSN

2019-05-29 11:44:26.493171+0900 [PRJ_NAME][309:8621] Connectionlogger: STARTTLS
2019-05-29 11:44:26.630403+0900 [PRJ_NAME][309:8621] Connectionlogger: 220 2.0.0 Ready to start TLS
2019-05-29 11:44:27.294594+0900 [PRJ_NAME][309:8621] Connectionlogger: EHLO [iPhone_NAME]
2019-05-29 11:44:27.452163+0900 [PRJ_NAME][309:8621] Connectionlogger: 250-pv50p00im-zteg10021401.me.com
250-PIPELINING
250-SIZE 28311552
250-ETRN
250-AUTH LOGIN PLAIN ATOKEN
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 DSN

2019-05-29 11:44:27.458321+0900 [PRJ_NAME][309:8621] Connectionlogger: AUTH LOGIN
2019-05-29 11:44:27.610578+0900 [PRJ_NAME][309:8621] Connectionlogger: 334 VXNlcm5hbWU6
2019-05-29 11:44:27.610993+0900 [PRJ_NAME][309:8621] Connectionlogger: am9ibWFnaWM3QGljbG91ZC5jb20=
2019-05-29 11:44:27.750485+0900 [PRJ_NAME][309:8621] Connectionlogger: 334 UGFzc3dvcmQ6
2019-05-29 11:44:27.750916+0900 [PRJ_NAME][309:8621] Connectionlogger: aHlnay11eG9zLWRscmwteWNmYQ==
2019-05-29 11:44:28.048041+0900 [PRJ_NAME][309:8621] Connectionlogger: 235 2.7.0 Authentication successful (CKUSER)
2019-05-29 11:44:28.048344+0900 [PRJ_NAME][309:8621] Connectionlogger: MAIL FROM:<[ICLOUD_MAILADDRSS]> SIZE=755
2019-05-29 11:44:28.198642+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.1.0 Ok
2019-05-29 11:44:28.199076+0900 [PRJ_NAME][309:8621] Connectionlogger: RCPT TO:<[TO_ADDRESS]>
2019-05-29 11:44:28.359165+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.1.5 Ok
2019-05-29 11:44:28.359622+0900 [PRJ_NAME][309:8621] Connectionlogger: DATA
2019-05-29 11:44:28.520164+0900 [PRJ_NAME][309:8621] Connectionlogger: 354 End data with <CR><LF>.<CR><LF>
2019-05-29 11:44:28.521277+0900 [PRJ_NAME][309:8621] Connectionlogger: Date: Wed, 29 May 2019 11:44:24 +0900

From: JobMagic7 <[ICLOUD_MAILADDRSS]>
To: [TO_ADDRESS]
Message-ID: <ba79c67a-1c6a-4537-8307-36f5112adcb6@[iPhone_NAME]>

Subject: 
 =?utf-8?Q?=E3=83=86=E3=82=B9=E3=83=88=E3=83=A1=E3=83=BC=E3=83=AB1?=
MIME-Version: 1.0

Content-Type: multipart/alternative; boundary="5cedf208_6b8b4567_135"

--5cedf208_6b8b4567_135
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline

=E3=83=86=E3=82=B9=E3=83=88=E6=9C=AC=E6=96=87

--5cedf208_6b8b4567_135
Content-Type: text/html; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline

=E3=83=86=E3=82=B9=E3=83=88=E6=9C=AC=E6=96=87

--5cedf208_6b8b4567_135--

.

2019-05-29 11:44:28.522446+0900 [PRJ_NAME][309:8621] Connectionlogger: QUIT
2019-05-29 11:44:28.978250+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.0.0 Ok: queued as 4524F480C10
2019-05-29 11:44:28.980927+0900 [PRJ_NAME][309:8502] Successfully sent email!

**Comment by potetochippusu** [comment]
Here you will.
Is that caused by having the same iCloud email account settings on the device mailer?

[Source code]
-------------------------------------------------------------------
let smtpSession = MCOSMTPSession()

smtpSession.hostname = "smtp.mail.me.com"
smtpSession.username = "[ICLOUD_MAILADDRSS]"
smtpSession.password = "[AppPassword]"
smtpSession.port = 587
smtpSession.authType = MCOAuthType.saslLogin
smtpSession.connectionType = MCOConnectionType.startTLS

smtpSession.connectionLogger = {(connectionID, type, data) in
    if data != nil {
        if let string = NSString(data: data!, encoding: String.Encoding.utf8.rawValue){
            NSLog("Connectionlogger: \(string)")
        }
    }
}

let builder = MCOMessageBuilder()
builder.header.to = [MCOAddress(displayName: "", mailbox: "[TO_ADDRESS]")]
builder.header.from = MCOAddress(displayName: "", mailbox: "[ICLOUD_MAILADDRSS]")

builder.header.subject = "[MAIL_TITLE]"
builder.htmlBody = "[MAIL_BODY]"

let rfc822Data = builder.data()
let sendOperation = smtpSession.sendOperation(with: rfc822Data)
sendOperation!.start { (error) -> Void in
    if (error != nil) {
        NSLog("Error sending email: \(error)")
    } else {
        NSLog("Successfully sent email!")
    }
}
-------------------------------------------------------------------

[Send log]
-------------------------------------------------------------------
2019-05-29 11:44:26.349089+0900 [PRJ_NAME][309:8621] Connectionlogger: 220 pv50p00im-zteg10021401.me.com 3.3.1 v.666
2019-05-29 11:44:26.349777+0900 [PRJ_NAME][309:8621] Connectionlogger: EHLO [iPhone_NAME]
2019-05-29 11:44:26.492623+0900 [PRJ_NAME][309:8621] Connectionlogger: 250-pv50p00im-zteg10021401.me.com
250-PIPELINING
250-SIZE 28311552
250-ETRN
250-STARTTLS
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 DSN

2019-05-29 11:44:26.493171+0900 [PRJ_NAME][309:8621] Connectionlogger: STARTTLS
2019-05-29 11:44:26.630403+0900 [PRJ_NAME][309:8621] Connectionlogger: 220 2.0.0 Ready to start TLS
2019-05-29 11:44:27.294594+0900 [PRJ_NAME][309:8621] Connectionlogger: EHLO [iPhone_NAME]
2019-05-29 11:44:27.452163+0900 [PRJ_NAME][309:8621] Connectionlogger: 250-pv50p00im-zteg10021401.me.com
250-PIPELINING
250-SIZE 28311552
250-ETRN
250-AUTH LOGIN PLAIN ATOKEN
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 DSN

2019-05-29 11:44:27.458321+0900 [PRJ_NAME][309:8621] Connectionlogger: AUTH LOGIN
2019-05-29 11:44:27.610578+0900 [PRJ_NAME][309:8621] Connectionlogger: 334 VXNlcm5hbWU6
2019-05-29 11:44:27.610993+0900 [PRJ_NAME][309:8621] Connectionlogger: am9ibWFnaWM3QGljbG91ZC5jb20=
2019-05-29 11:44:27.750485+0900 [PRJ_NAME][309:8621] Connectionlogger: 334 UGFzc3dvcmQ6
2019-05-29 11:44:27.750916+0900 [PRJ_NAME][309:8621] Connectionlogger: aHlnay11eG9zLWRscmwteWNmYQ==
2019-05-29 11:44:28.048041+0900 [PRJ_NAME][309:8621] Connectionlogger: 235 2.7.0 Authentication successful (CKUSER)
2019-05-29 11:44:28.048344+0900 [PRJ_NAME][309:8621] Connectionlogger: MAIL FROM:<[ICLOUD_MAILADDRSS]> SIZE=755
2019-05-29 11:44:28.198642+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.1.0 Ok
2019-05-29 11:44:28.199076+0900 [PRJ_NAME][309:8621] Connectionlogger: RCPT TO:<[TO_ADDRESS]>
2019-05-29 11:44:28.359165+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.1.5 Ok
2019-05-29 11:44:28.359622+0900 [PRJ_NAME][309:8621] Connectionlogger: DATA
2019-05-29 11:44:28.520164+0900 [PRJ_NAME][309:8621] Connectionlogger: 354 End data with <CR><LF>.<CR><LF>
2019-05-29 11:44:28.521277+0900 [PRJ_NAME][309:8621] Connectionlogger: Date: Wed, 29 May 2019 11:44:24 +0900

From: [ICLOUD_MAILADDRSS]
To: [TO_ADDRESS]
Message-ID: <ba79c67a-1c6a-4537-8307-36f5112adcb6@[iPhone_NAME]>

Subject: 
 =?utf-8?Q?=E3=83=86=E3=82=B9=E3=83=88=E3=83=A1=E3=83=BC=E3=83=AB1?=
MIME-Version: 1.0

Content-Type: multipart/alternative; boundary="5cedf208_6b8b4567_135"

--5cedf208_6b8b4567_135
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline

=E3=83=86=E3=82=B9=E3=83=88=E6=9C=AC=E6=96=87

--5cedf208_6b8b4567_135
Content-Type: text/html; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline

=E3=83=86=E3=82=B9=E3=83=88=E6=9C=AC=E6=96=87

--5cedf208_6b8b4567_135--

.

2019-05-29 11:44:28.522446+0900 [PRJ_NAME][309:8621] Connectionlogger: QUIT
2019-05-29 11:44:28.978250+0900 [PRJ_NAME][309:8621] Connectionlogger: 250 2.0.0 Ok: queued as 4524F480C10
2019-05-29 11:44:28.980927+0900 [PRJ_NAME][309:8502] Successfully sent email!
-------------------------------------------------------------------

**Comment by haithngn** [comment]
It seems iCloud SMTP block your IP, could you try send with another internet network ?

**Comment by potetochippusu** [comment]
Thank you for your response.

I tried with two types of internet access, but both did not reach the destination email address.

Am I missing something on the iPhone or icloud?

--------------------------------------------------------------------
[What I know now]

・ When the account of From is icloud.com
-> Execute in the simulator: It will reach the destination email address
-> Execute on iPhone: Do not reach destination email address

・ When account of From is other than icloud.com
-> Execute in the simulator: It will reach the destination email address
-> Execute on iPhone: Receive to the destination email address

Transmission log
-> No error
-> The difference between running on a simulator and an iPhone terminal is that the line feed is increased and output when executed on an iPhone terminal

・ Change of line
-> I tried two types but both failed

・ Error mail
-> None

・ When using iPhone mailer
-> I successfully sent / received by icloud.com account email.

**Comment by potetochippusu** [comment]
It is an additional progress report.

I did an internal test with TestFlight, but the results were the same.
It was a failure.

**Comment by goburin** [comment]
Hi!
I also want to use MailCore2.

I was interested in this issue, so I examined it.

I got the same result.
I tried with the iPhone XS.

Is there anyone who successfully sent SMTP from icloud email account using iPhone?

**Comment by potetochippusu** [comment]
Hello.
Long time no see.

After that, I checked variously and understood the following.

The cause seems to be the problem in the following part of [Send log] posted the other day.
-------------------------------------------------------- -
Connectionlogger: EHLO [iPhone_NAME]
-------------------------------------------------------- -

In the above log, [iPhone_NAME] part was displayed as [Mac_Name] .local when executed in the simulator.

I have to be displayed as [iPhone_NAME] .local (host name) on the actual iPhone, but I can not get the host name.
It will only be [iPhone_NAME].

Is this a problem with setting up the iPhone?
Do I need to modify the library source myself?

If you know, can you give me some advice?

**Comment by potetochippusu** [comment]
I will report in addition.

1. The result has been changed by the information of the destination.

  If the destination address is one, no email will be sent if the displayName is empty.
  (E-mail arrived when you set some character in displayName.)

  If there are two or more destination addresses, the email has been sent even if the displayName is empty.

2. I tried to contact Apple.

  There was an answer to comply with RFC 5321 EHLO command.


・Why does the result differ depending on the presence or absence of displayName?
・How can I check if it is compliant with RFC 5321?

**Comment by triton3** [comment]
@potetochippusu How did you solve this problem? I am seeing the same issue on @me.com account, but it does not seem to be related to display name at all, and neither with the number of recipients. The SMTP server does not show any error . The connection logs I see are the same as what you reported.

**Comment by triton3** [comment]
Just leaving a note in case others see this issue. In our case, we found that while emails sent from @me.com or @icloud.com from one iPhone weren't being delivered, on another device they worked fine.

We did a reset on the device name for the iPhone that didn't work (changed the name in _Settings -> General -> About -> Name_) from previous value to just "iPhone", and tested sending an email. It was then delivered correctly. After that, restoring the device name to its previous value also worked and all emails were being delivered.
